### ISSUE TITLE
Bump Gradle Wrapper to v8.4

### ISSUE THREAD

**Issue by JoseLion** [microsoft/vscode-gradle#1420]
**Is your feature request related to a problem? Please describe.**
It'd be great to have support for JDK21 as it was recently released as the LTS for the next 2 years. The extension works great with JDK20, but on 21 it throws the error below:

<details>
<summary><i>Gradle for Java (Language Server) - Output</i></summary>

```console
Sep 21, 2023 11:28:15 PM org.eclipse.lsp4j.jsonrpc.RemoteEndpoint fallbackResponseError
SEVERE: Internal error: java.lang.reflect.InvocationTargetException
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 11 more
Caused by: BUG! exception in phase 'semantic analysis' in source unit 'file:///c%3A/Repos/atomic-flyway/build.gradle' Unsupported class file major version 65
	at org.codehaus.groovy.control.CompilationUnit$ISourceUnitOperation.doPhaseOperation(CompilationUnit.java:975)
	at org.codehaus.groovy.control.CompilationUnit.processPhaseOperations(CompilationUnit.java:710)
	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:684)
	at com.microsoft.gradle.GradleServices.compile(GradleServices.java:178)
	at com.microsoft.gradle.GradleServices.recompileAll(GradleServices.java:195)
	at com.microsoft.gradle.GradleServices.executeCommand(GradleServices.java:379)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	... 13 more
Caused by: java.lang.IllegalArgumentException: Unsupported class file major version 65
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:199)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:180)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:166)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:287)
	at org.codehaus.groovy.ast.decompiled.AsmDecompiler.parseClass(AsmDecompiler.java:83)
	at org.codehaus.groovy.control.ClassNodeResolver.findDecompiled(ClassNodeResolver.java:255)
	at org.codehaus.groovy.control.ClassNodeResolver.tryAsLoaderClassOrScript(ClassNodeResolver.java:193)
	at org.codehaus.groovy.control.ClassNodeResolver.findClassNode(ClassNodeResolver.java:175)
	at org.codehaus.groovy.control.ClassNodeResolver.resolveName(ClassNodeResolver.java:129)
	at org.codehaus.groovy.control.ResolveVisitor.resolveToOuter(ResolveVisitor.java:879)
	at org.codehaus.groovy.control.ResolveVisitor.resolve(ResolveVisitor.java:469)
	at org.codehaus.groovy.control.ResolveVisitor.resolveFromDefaultImports(ResolveVisitor.java:635)
	at org.codehaus.groovy.control.ResolveVisitor.resolveFromDefaultImports(ResolveVisitor.java:599)
	at org.codehaus.groovy.control.ResolveVisitor.resolve(ResolveVisitor.java:468)
	at org.codehaus.groovy.control.ResolveVisitor.resolve(ResolveVisitor.java:430)
	at org.codehaus.groovy.control.ResolveVisitor.transformVariableExpression(ResolveVisitor.java:1151)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:898)
	at org.codehaus.groovy.control.ResolveVisitor.transformPropertyExpression(ResolveVisitor.java:1023)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:900)
	at org.codehaus.groovy.control.ResolveVisitor.transformBinaryExpression(ResolveVisitor.java:1176)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:904)
	at org.codehaus.groovy.ast.expr.Expression.transformExpressions(Expression.java:49)
	at org.codehaus.groovy.ast.expr.GStringExpression.transformExpression(GStringExpression.java:61)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:915)
	at org.codehaus.groovy.ast.expr.MapEntryExpression.transformExpression(MapEntryExpression.java:42)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:915)
	at org.codehaus.groovy.ast.expr.Expression.transformExpressions(Expression.java:63)
	at org.codehaus.groovy.ast.expr.NamedArgumentListExpression.transformExpression(NamedArgumentListExpression.java:38)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:915)
	at org.codehaus.groovy.ast.expr.Expression.transformExpressions(Expression.java:49)
	at org.codehaus.groovy.ast.expr.TupleExpression.transformExpression(TupleExpression.java:96)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:915)
	at org.codehaus.groovy.control.ResolveVisitor.transformMethodCallExpression(ResolveVisitor.java:1268)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:906)
	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitExpressionStatement(ClassCodeExpressionTransformer.java:129)
	at org.codehaus.groovy.ast.stmt.ExpressionStatement.visit(ExpressionStatement.java:41)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:86)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:208)
	at org.codehaus.groovy.control.ResolveVisitor.visitBlockStatement(ResolveVisitor.java:1564)
	at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:70)
	at org.codehaus.groovy.control.ResolveVisitor.transformClosureExpression(ResolveVisitor.java:1246)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:908)
	at org.codehaus.groovy.ast.expr.Expression.transformExpressions(Expression.java:49)
	at org.codehaus.groovy.ast.expr.ArgumentListExpression.transformExpression(ArgumentListExpression.java:67)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:915)
	at org.codehaus.groovy.control.ResolveVisitor.transformMethodCallExpression(ResolveVisitor.java:1268)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:906)
	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitExpressionStatement(ClassCodeExpressionTransformer.java:129)
	at org.codehaus.groovy.ast.stmt.ExpressionStatement.visit(ExpressionStatement.java:41)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:86)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:208)
	at org.codehaus.groovy.control.ResolveVisitor.visitBlockStatement(ResolveVisitor.java:1564)
	at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:70)
	at org.codehaus.groovy.control.ResolveVisitor.transformClosureExpression(ResolveVisitor.java:1246)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:908)
	at org.codehaus.groovy.ast.expr.Expression.transformExpressions(Expression.java:49)
	at org.codehaus.groovy.ast.expr.ArgumentListExpression.transformExpression(ArgumentListExpression.java:67)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:915)
	at org.codehaus.groovy.control.ResolveVisitor.transformMethodCallExpression(ResolveVisitor.java:1268)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:906)
	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitExpressionStatement(ClassCodeExpressionTransformer.java:129)
	at org.codehaus.groovy.ast.stmt.ExpressionStatement.visit(ExpressionStatement.java:41)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:86)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:208)
	at org.codehaus.groovy.control.ResolveVisitor.visitBlockStatement(ResolveVisitor.java:1564)
	at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:70)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitClassCodeContainer(ClassCodeVisitorSupport.java:157)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitConstructorOrMethod(ClassCodeVisitorSupport.java:130)
	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitConstructorOrMethod(ClassCodeExpressionTransformer.java:87)
	at org.codehaus.groovy.control.ResolveVisitor.visitConstructorOrMethod(ResolveVisitor.java:315)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitMethod(ClassCodeVisitorSupport.java:125)
	at org.codehaus.groovy.control.ResolveVisitor.visitMethod(ResolveVisitor.java:285)
	at org.codehaus.groovy.ast.ClassNode.visitMethods(ClassNode.java:1191)
	at org.codehaus.groovy.ast.ClassNode.visitContents(ClassNode.java:1184)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitClass(ClassCodeVisitorSupport.java:56)
	at org.codehaus.groovy.control.ResolveVisitor.visitClass(ResolveVisitor.java:1480)
	at org.codehaus.groovy.control.ResolveVisitor.startResolving(ResolveVisitor.java:266)
	at org.codehaus.groovy.control.CompilationUnit.lambda$4(CompilationUnit.java:229)
	at org.codehaus.groovy.control.CompilationUnit$ISourceUnitOperation.doPhaseOperation(CompilationUnit.java:971)
	... 19 more

[Error - 11:28:15 PM] Request workspace/executeCommand failed.
  Message: Internal error.
  Code: -32603 
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:67)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.request(GenericEndpoint.java:120)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleRequest(RemoteEndpoint.java:261)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:190)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.eclipse.lsp4j.jsonrpc.services.GenericEndpoint.lambda$null$0(GenericEndpoint.java:65)
	... 11 more
Caused by: BUG! exception in phase 'semantic analysis' in source unit 'file:///c%3A/Repos/atomic-flyway/build.gradle' Unsupported class file major version 65
	at org.codehaus.groovy.control.CompilationUnit$ISourceUnitOperation.doPhaseOperation(CompilationUnit.java:975)
	at org.codehaus.groovy.control.CompilationUnit.processPhaseOperations(CompilationUnit.java:710)
	at org.codehaus.groovy.control.CompilationUnit.compile(CompilationUnit.java:684)
	at com.microsoft.gradle.GradleServices.compile(GradleServices.java:178)
	at com.microsoft.gradle.GradleServices.recompileAll(GradleServices.java:195)
	at com.microsoft.gradle.GradleServices.executeCommand(GradleServices.java:379)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	... 13 more
Caused by: java.lang.IllegalArgumentException: Unsupported class file major version 65
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:199)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:180)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:166)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:287)
	at org.codehaus.groovy.ast.decompiled.AsmDecompiler.parseClass(AsmDecompiler.java:83)
	at org.codehaus.groovy.control.ClassNodeResolver.findDecompiled(ClassNodeResolver.java:255)
	at org.codehaus.groovy.control.ClassNodeResolver.tryAsLoaderClassOrScript(ClassNodeResolver.java:193)
	at org.codehaus.groovy.control.ClassNodeResolver.findClassNode(ClassNodeResolver.java:175)
	at org.codehaus.groovy.control.ClassNodeResolver.resolveName(ClassNodeResolver.java:129)
	at org.codehaus.groovy.control.ResolveVisitor.resolveToOuter(ResolveVisitor.java:879)
	at org.codehaus.groovy.control.ResolveVisitor.resolve(ResolveVisitor.java:469)
	at org.codehaus.groovy.control.ResolveVisitor.resolveFromDefaultImports(ResolveVisitor.java:635)
	at org.codehaus.groovy.control.ResolveVisitor.resolveFromDefaultImports(ResolveVisitor.java:599)
	at org.codehaus.groovy.control.ResolveVisitor.resolve(ResolveVisitor.java:468)
	at org.codehaus.groovy.control.ResolveVisitor.resolve(ResolveVisitor.java:430)
	at org.codehaus.groovy.control.ResolveVisitor.transformVariableExpression(ResolveVisitor.java:1151)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:898)
	at org.codehaus.groovy.control.ResolveVisitor.transformPropertyExpression(ResolveVisitor.java:1023)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:900)
	at org.codehaus.groovy.control.ResolveVisitor.transformBinaryExpression(ResolveVisitor.java:1176)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:904)
	at org.codehaus.groovy.ast.expr.Expression.transformExpressions(Expression.java:49)
	at org.codehaus.groovy.ast.expr.GStringExpression.transformExpression(GStringExpression.java:61)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:915)
	at org.codehaus.groovy.ast.expr.MapEntryExpression.transformExpression(MapEntryExpression.java:42)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:915)
	at org.codehaus.groovy.ast.expr.Expression.transformExpressions(Expression.java:63)
	at org.codehaus.groovy.ast.expr.NamedArgumentListExpression.transformExpression(NamedArgumentListExpression.java:38)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:915)
	at org.codehaus.groovy.ast.expr.Expression.transformExpressions(Expression.java:49)
	at org.codehaus.groovy.ast.expr.TupleExpression.transformExpression(TupleExpression.java:96)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:915)
	at org.codehaus.groovy.control.ResolveVisitor.transformMethodCallExpression(ResolveVisitor.java:1268)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:906)
	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitExpressionStatement(ClassCodeExpressionTransformer.java:129)
	at org.codehaus.groovy.ast.stmt.ExpressionStatement.visit(ExpressionStatement.java:41)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:86)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:208)
	at org.codehaus.groovy.control.ResolveVisitor.visitBlockStatement(ResolveVisitor.java:1564)
	at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:70)
	at org.codehaus.groovy.control.ResolveVisitor.transformClosureExpression(ResolveVisitor.java:1246)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:908)
	at org.codehaus.groovy.ast.expr.Expression.transformExpressions(Expression.java:49)
	at org.codehaus.groovy.ast.expr.ArgumentListExpression.transformExpression(ArgumentListExpression.java:67)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:915)
	at org.codehaus.groovy.control.ResolveVisitor.transformMethodCallExpression(ResolveVisitor.java:1268)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:906)
	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitExpressionStatement(ClassCodeExpressionTransformer.java:129)
	at org.codehaus.groovy.ast.stmt.ExpressionStatement.visit(ExpressionStatement.java:41)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:86)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:208)
	at org.codehaus.groovy.control.ResolveVisitor.visitBlockStatement(ResolveVisitor.java:1564)
	at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:70)
	at org.codehaus.groovy.control.ResolveVisitor.transformClosureExpression(ResolveVisitor.java:1246)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:908)
	at org.codehaus.groovy.ast.expr.Expression.transformExpressions(Expression.java:49)
	at org.codehaus.groovy.ast.expr.ArgumentListExpression.transformExpression(ArgumentListExpression.java:67)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:915)
	at org.codehaus.groovy.control.ResolveVisitor.transformMethodCallExpression(ResolveVisitor.java:1268)
	at org.codehaus.groovy.control.ResolveVisitor.transform(ResolveVisitor.java:906)
	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitExpressionStatement(ClassCodeExpressionTransformer.java:129)
	at org.codehaus.groovy.ast.stmt.ExpressionStatement.visit(ExpressionStatement.java:41)
	at org.codehaus.groovy.ast.CodeVisitorSupport.visitBlockStatement(CodeVisitorSupport.java:86)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitBlockStatement(ClassCodeVisitorSupport.java:208)
	at org.codehaus.groovy.control.ResolveVisitor.visitBlockStatement(ResolveVisitor.java:1564)
	at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:70)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitClassCodeContainer(ClassCodeVisitorSupport.java:157)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitConstructorOrMethod(ClassCodeVisitorSupport.java:130)
	at org.codehaus.groovy.ast.ClassCodeExpressionTransformer.visitConstructorOrMethod(ClassCodeExpressionTransformer.java:87)
	at org.codehaus.groovy.control.ResolveVisitor.visitConstructorOrMethod(ResolveVisitor.java:315)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitMethod(ClassCodeVisitorSupport.java:125)
	at org.codehaus.groovy.control.ResolveVisitor.visitMethod(ResolveVisitor.java:285)
	at org.codehaus.groovy.ast.ClassNode.visitMethods(ClassNode.java:1191)
	at org.codehaus.groovy.ast.ClassNode.visitContents(ClassNode.java:1184)
	at org.codehaus.groovy.ast.ClassCodeVisitorSupport.visitClass(ClassCodeVisitorSupport.java:56)
	at org.codehaus.groovy.control.ResolveVisitor.visitClass(ResolveVisitor.java:1480)
	at org.codehaus.groovy.control.ResolveVisitor.startResolving(ResolveVisitor.java:266)
	at org.codehaus.groovy.control.CompilationUnit.lambda$4(CompilationUnit.java:229)
	at org.codehaus.groovy.control.CompilationUnit$ISourceUnitOperation.doPhaseOperation(CompilationUnit.java:971)
	... 19 more
```
</details>

**Describe the solution you'd like**
The extension works with JDK21 just as with previous JDK versions.

**Describe alternatives you've considered**
I don't see any other alternatives if we want to work with JDK21 on Gradle projects.

**Additional context**
I'm currently using the pre-release version `v3.12.2023091801` of the extension along with `java.gradle.buildServer.enabled` setting set to `on`. Another thing to consider is that Gradle only supports Java 21 from `v8.4-rc-1`, but from the looks of their [GitHub Milestone](https://github.com/gradle/gradle/milestone/244), only some documentation changes are needed for `v8.4` to be complete.

<details>
<summary><i>Gradle For Java - Output</i></summary>

```console
[info] Build file changed: c:\Repos\atomic-flyway\build.gradle
[info] Java Home: C:\Program Files\Java\jdk-21
[info] JVM Args: --add-opens=java.base/java.util=ALL-UNNAMED,--add-opens=java.base/java.lang=ALL-UNNAMED,--add-opens=java.base/java.lang.invoke=ALL-UNNAMED,--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED,--add-opens=java.base/java.nio.charset=ALL-UNNAMED,--add-opens=java.base/java.net=ALL-UNNAMED,--add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED,-XX:MaxMetaspaceSize=384m,-XX:+HeapDumpOnOutOfMemoryError,-Xms256m,-Xmx512m,-Dfile.encoding=UTF-8,-Duser.country=US,-Duser.language=en,-Duser.variant
[info] Gradle User Home: C:\Users\josel\.gradle
[info] Gradle Version: 8.4-rc-1
[info] Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.4-rc-1/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

CONFIGURE SUCCESSFUL in 239ms
[info] Found 56 tasks
```
</details

**Comment by robaho** [comment]
The extension is not working with the released Gradle 8.4.

Forward Java support should be a core requirement of the extension. The ast parsing should be a function of the running JVM not a library as to supported formats.

**Comment by jdneo** [comment]
> The extension is not working with the released Gradle 8.4.

Can you give more details about how it is not working with Gradle 8.4?

**Comment by robaho** [comment]
> > The extension is not working with the released Gradle 8.4.
> 
> Can you give more details about how it is not working with Gradle 8.4?

It is the same error as the OP. Gradle's recommendation is to run Gradle using JDK-20, even when compiling/running for JDK-21. When you do that, I think the extension picks up the Gradle JVM of JDK-20 and so can't read the compiled classes (which were compiled with JDK-21).

i.e. this error

```java
Caused by: java.lang.IllegalArgumentException: Unsupported class file major version 65
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:199)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:180)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:166)
	at groovyjarjarasm.asm.ClassReader.<init>(ClassReader.java:287)
	at org.codehaus.groovy.ast.decompiled.AsmDecompiler.parseClass(AsmDecompiler.java:83)
	at org.codehaus.groovy.control.ClassNodeResolver.findDecompiled(ClassNodeResolver.java:255)
	at org.codehaus.groovy.control.ClassNodeResolver.tryAsLoaderClassOrScript(ClassNodeResolver.java:193)
	at org.codehaus.groovy.control.ClassNodeResolver.findClassNode(ClassNodeResolver.java:175)
	at org.codehaus.groovy.control.ClassNodeResolver.resolveName(ClassNodeResolver.java:129)
	at org.codehaus.groovy.control.ResolveVisitor.resolveToOuter(ResolveVisitor.java:879)
	at org.codehaus.groovy.control.ResolveVisitor.resolve(ResolveVisitor.java:469)
```

**Comment by jdneo** [comment]
@robaho Got it!

Yes, it's related with the implementation of build server for gradle. I will fix it ASAP.
